.changes/unreleased/Fixed-20241203-071352.yaml

```diff
@@ -0,0 +1,9 @@
+kind: Fixed
+body: |-
+    Ensure `RepoRootForImportDynamic` properly trims trailing `../` in ref
+
+    Fixes https://github.com/dagger/dagger/issues/9091
```

Comment:
`RepoRootForImportDynamic` isn't an exposed function - so we shouldn't include it in the changelog. Changelogs are for end-users, so we need to phrase this as an end-user fix.

Does this even need a changelog? From my understanding, we're only fixing an error case - the only refs this resolves weirdly now are refs that were actually invalid.

Reply:
Ahah, the only time I actually remember to add one it's not necessary 🤣